frontend/src/components/rule-edit/generation-rule/GenerationRuleListItem.vue

```diff
@@ -41,6 +41,7 @@ const handleDeleteClicked = () => {
             v-if="
               f_params !== null &&
               f_params !== false &&
+              f_params !== undefined &&
```

Comment:
Two things:
- testing for a variable being undefined in JS is usually done with `typeof variable == "undefined"`, I'm not sure why it is better than comparing to `undefined` directly like you do but maybe there are corner cases
- it feels like this whole test could be summed up as testing whether `f_params` is *truthy*, in JS there are several ways of converting a variable to a boolean, the one I've seen most is `!!variable` (two exclamation points). Would that work? I think we'd still need to test for empty array, but the three others should be covered.

Reply:
!!variable works perfectly. thanks